syscalls.go

```diff
@@ -357,10 +357,9 @@ func wrapObjError(err error) error {
 		return nil
 	}
 	if errors.Is(err, unix.ENOENT) {
-		return fmt.Errorf("%w", ErrNotExist)
```

Comment:
This is on purpose, it forces `errors.Is` rather than directly comparing the error.

Reply:
Should `wrapMapError` be doing that as well then? I was just mirroring what `wrapMapError` does.

Reply:
Yeah that is kind of bogus. I think in practice callers of wrapMapError add context and so provide the necessary wrapping, but really this function should wrap explicitly.

Seeing as you're adding more and more of these escape hatches I think I need to come up with a better way of dealing with syscall errors. For now, can you ensure that wrapObjErr does wrapping? I'll make a PR with a better long-term solution.

Reply:
Will do, thanks!